### PR TITLE
Add TiFlash test cases

### PR DESCRIPTION
**Pull request by YuJuncen**
### What problem does this PR solve? 
Currently, we have no test cases with TiFlash, this PR add TiFlash to test cluster.

And, at post, we should remove all TiFlash replicas so that we backup or restore successfully without failed by 'RPC unimplemented', this PR also add a flag to disable this function, for farther test usage.  

### What is changed and how it works?
I replace `store.GetAddr()` to `store.GetPeerAddr() || store.GetAddr()`, because TiFlash use a 'proxy' to act like a TiKV, its 'address' is more like a 'service' address but a 'peer' address. Hence, we should try get `PeerAddr` firstly, and normal TiKV has no `PeerAddr`, so we use `Addr` as a fallback.

And I added a one-node TiFlash to the integration test cluster.

### Check List 

Tests 

 - Integration test
 + Manual test (add detailed scripts or steps below)
    - Start a cluster with TiFlash(`tiup playground --tiflash 1`)
    - Run br without removing TiFlash replicas.
    - Restore success.

### Release Note

 - Fix a bug that might cause BR fail when operating with clusters with TiFlash

### CODE REVIEW

**Comment by YuJuncen** [comment]
/run-integration-test

It freezes at enabling placement rules...?

**Comment by YuJuncen** [comment]
`git@github.com: Permission denied (publickey)`

……🤔

**Comment by YuJuncen** [comment]
`/home/jenkins/agent/workspace/br_ghpr_unit_and_integration_test/go/src/github.com/pingcap/br/tests/_utils/run_services: line 118: bin/tiflash: No such file or directory`

Seems CI doesn't run `tests/install_tools.sh`, perhaps we have to add downloading TiFlash into CI pipeline...

**Comment by YuJuncen** [comment]
/run-integration-test

Is CI OK?

**Comment by kennytm** [comment]
```
open /home/jenkins/agent/workspace/br_ghpr_unit_and_integration_test/go/src/github.com/pingcap/br/tests/br_incompatible_tidb_config/config/pd.toml: no such file or directory
```

**Comment by YuJuncen** [comment]
/run-integration-test

That was strange... Why just one test case failed with `pd.toml` not found?

**Comment by YuJuncen** [comment]
/run-integration-test

Some of TiKV failed to start...?

`[2020-06-02T02:50:05.406Z] [2020/06/02 10:48:42.485 +08:00] [ERROR] [tso.go:126] ["system time may be incorrect"] [last=2020/06/02 10:48:45.093 +08:00] [next=2020/06/02 10:48:42.485 +08:00]
`

Maybe container is suspended...?

Replay successes: https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/br_ghpr_unit_and_integration_test/detail/br_ghpr_unit_and_integration_test/1801/pipeline/

**Comment by YuJuncen** [comment]
/run-integration-test

Seems set lease time and save TSO time to 360s works?

File: tests/_utils/run_services [hunk 1]
```diff
@@ -111,6 +122,25 @@ start_services() {
...
 }
 
+start_tiflash() {
+    echo "Starting TiFlash..."
+    LD_LIBRARY_PATH=bin/ bin/tiflash server --config-file=$TIFLASH_CONFIG &
+    echo "TiFlash started..."
+
+    i=0
+    # FIXME: Add retry to makesure TiFlash start here, currently, seems pd-ctl NEVER returns at this context.
+    while false # ! bin/pd-ctl store | grep "tiflash"
```
Comment (overvenus) [inline tests/_utils/run_services:132]: We can curl tiflash status page (it's the page from an embeded tikv in tiflash).

https://github.com/pingcap/ticdc/blob/33f568e6556e047ab6dacb42d4ac0ce5ef7b53a3/tests/_utils/start_tidb_cluster#L250

Comment (YuJuncen) [reply tests/_utils/run_services]: That's a smart solution.

BTW, I copied the way CDC making TiFlash configuration files :smile:.

**Comment by YuJuncen** [comment]
/run-integration-test

Let's see whether test cases are stable enough.

**Comment by YuJuncen** [comment]
Should we take this PR to 3.1? @overvenus

File: tests/br_tiflash/run.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+run_sql "ALTER TABLE $DB.kv SET TIFLASH REPLICA 1" 
+
+stmt="INSERT INTO $DB.kv(k, v) VALUES ('1-record', 1)"
+for i in $(seq 2 $RECORD_COUNT); do
+    stmt="$stmt,('$i-record', $i)"
+done
+run_sql "$stmt"
+
+rm -rf "/${TEST_DIR}/$DB"
+run_br backup full -s "local://$TEST_DIR/$DB" --pd $PD_ADDR
```
Comment (overvenus) [inline tests/br_tiflash/run.sh:33]: Could you check tiflash replication progress before backup? 

https://pingcap.com/docs/stable/tiflash/use-tiflash/#check-the-replication-progress

**Comment by overvenus** [comment]
> Should we take this PR to 3.1? @overvenus

Yes, please, as tiflash also supports v3.1.x

File: tests/br_tiflash/run.sh (hunk 2 above)
Comment (YuJuncen) [reply tests/br_tiflash/run.sh:33]: This part is buggy now... Let me check my config...

Comment (YuJuncen) [reply tests/br_tiflash/run.sh:33]:
```
Traceback (most recent call last):
  File "flash_cluster_manager.py", line 261, in main
  File "flash_cluster_manager.py", line 139, in __init__
  File "flash_cluster_manager.py", line 152, in _update_cluster
  File "flash_cluster_manager.py", line 79, in compute_cur_store
Exception: Can not tell current store.
service_addr: 127.0.0.1:3932,
all tiflash stores: [{'id': 105, 'address': '127.0.0.1:3930', 'labels': [{'key': 'engine', 'value': 'tiflash'}], 'version': 'v4.1.0-alpha-48-g482f94ae6', 'peer_address': '127.0.0.1:20170', 'status_address': '127.0.0.1:17000', 'git_hash': '482f94ae6b96d282a1ecc5c4d3e84ae226f6e7d4', 'start_timestamp': 1591085657, 'deploy_path': '/home/hillium/GolandProjects/br/bin', 'last_heartbeat': 1591085667839402817, 'state_name': 'Up'}]
```
Why...? :thinking:

Comment (YuJuncen) [reply tests/br_tiflash/run.sh:33]: 3932 vs 3930 ...:see_no_evil:

**Comment by YuJuncen** [comment]
/run-integration-test

`[2020-06-02T09:27:40.974Z] ERROR 9005 (HY000) at line 1: Region is unavailable`
Placement rules make things bad?

**Pull request merged**